packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -421,17 +429,40 @@ export class FluidDataStoreRuntime
 		this.localChangesTelemetryCount =
 			this.mc.config.getNumber("Fluid.Telemetry.LocalChangesTelemetryCount") ?? 10;
 
-		// eslint-disable-next-line import/no-deprecated
-		const base: IContainerRuntimeBaseExperimental | undefined =
+		// Reference these properties to avoid unused private member errors.
+		// They're accessed via IFluidDataStoreRuntimeExperimental interface.
+		((...experimental) => {})([this.inStagingMode, this.isDirty]);
+	}
+
+	/**
+	 * Implementation of IFluidDataStoreRuntimeExperimental.inStagingMode
+	 */
+	// eslint-disable-next-line import/no-deprecated
+	private get inStagingMode(): IFluidDataStoreRuntimeExperimental["inStagingMode"] {
+		return (
 			// eslint-disable-next-line import/no-deprecated
-			this.dataStoreContext.containerRuntime satisfies IContainerRuntimeBaseExperimental;
-		if (base !== undefined && "inStagingMode" in base) {
-			Object.defineProperty(this, "inStagingMode", {
-				get: () => {
-					return base.inStagingMode;
-				},
-			});
+			(this.dataStoreContext.containerRuntime as IContainerRuntimeBaseExperimental)
+				.inStagingMode
+		);
+	}
+
+	/**
+	 * Implementation of IFluidDataStoreRuntimeExperimental.isDirty
+	 */
+	// eslint-disable-next-line import/no-deprecated
+	private get isDirty(): IFluidDataStoreRuntimeExperimental["isDirty"] {
+		const dirty = this.pendingOpCount > 0;
+
+		const containerIsDirty =
+			// eslint-disable-next-line import/no-deprecated
+			(this.dataStoreContext.containerRuntime as IContainerRuntimeBaseExperimental).isDirty;
+		if (containerIsDirty === false && this.id !== agentSchedulerId) {
```

Comment:
i don't know if this matter very much. it is a bit awkward if the agent scheduler ds is dirty but the container isn't, but it also doesn't seem very common, so i'd probably just do nothing, or see if we can isolate the change in the agent scheduler dds itself